### PR TITLE
Updates Sunpy core with package template

### PR DESCRIPTION
**Pull request by CyclingNinja**
* Adds `.cruft.json` 
* General editing and compliance with new package template

### CODE REVIEW

File: .ruff.toml [hunk 1]
```diff
@@ -25,12 +25,14 @@ extend-ignore = [
...
-[per-file-ignores]
+[lint.per-file-ignores]
 # Part of configuration, not a package.
 "setup.py" = ["INP001"]
 "conftest.py" = ["INP001"]
-# Implicit-namespace-package. The examples are not a package.
-"docs/*.py" = ["INP001"]
+"docs/*.py" = [
+  "INP001",  # Implicit-namespace-package. The examples are not a package.
+  "E402",  # Module imports not at top of file
```
Comment (Cadair) [inline .ruff.toml:34]: This rule should only apply to conf.py

**Review by Cadair — APPROVED** [review]

**Comment by CyclingNinja** [comment]
pre-commit.ci autofix

**Comment by CyclingNinja** [comment]
pre-commit.ci autofox

**Review by wtbarnes — APPROVED** [review]
I am not the most qualified to review this, but this all looks good to me.

**Comment by Cadair** [comment]
I think the remaining todo here is that the oldestdeps build is broken.

**Review by nabobalis — APPROVED** [review]

**Review by nabobalis — CHANGES_REQUESTED** [review]
blocking until oldest deps is fixed

**Comment by Cadair** [comment]
The oldestdeps build is also failing on main, so I think that this is good to go.

File: sunpy/tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+"""
```
Comment (nabobalis) [inline sunpy/tests/__init__.py:1]: Does this need to be here?

**Review by nabobalis — APPROVED** [review]

**Pull request merged**